airbyte-server/src/main/java/io/airbyte/server/handlers/ConnectionsHandler.java

```diff
@@ -545,8 +550,9 @@ public boolean matchSearch(final DestinationSearch destinationSearch, final Dest
     return (destinationReadFromSearch == null || destinationReadFromSearch.equals(destinationRead));
   }
 
-  public void deleteConnection(final UUID connectionId) {
-    eventRunner.deleteConnection(connectionId);
+  public void deleteConnection(final UUID connectionId) throws JsonValidationException, ConfigNotFoundException, IOException {
+    connectionHelper.deleteConnection(connectionId);
```

Comment:
can we end up in the state where the connection gets "deleted" in the db but the temporal workflow doesn't get killed? or if that does happen does it get remedied automatically somehow?

Reply:
If the signal got send we temporal ensure that it will be run so we shouldn't end up in this use case.